arch/arm64/src/common/arm64_head.S

```diff
@@ -235,6 +235,15 @@ switch_el:
 
     bl    arm64_boot_el3_init
 
+#ifdef CONFIG_ARCH_BOOT_EL3
+    msr   SPSel, #1
+
+    /* Set SP_EL3 (with SPSel = 1) */
+
+    mov   sp, x24
+    b     el3_boot
+#endif
```

Comment:
If CONFIG_ARCH_BOOT_EL3 is defined, do you ever need the lines below? I.e. could this be #else here and #endif before the isb below?

Reply:
Right, could save a few lines

Reply:
Yes, the macros look a bit confusing. If I understand this correctly, the original code assumes that if "CONFIG_ARCH_HAVE_EL3" is enabled, it expects to enter here in EL3 and then switch to EL1. And now if "CONFIG_ARCH_BOOT_EL3" is defined, it continues booting in EL3

If I did get this right, then you can as well just #else away the EL switching if "CONFIG_ARCH_BOOT_EL3" is defined. It doesn't change the functionality though, IMHO.

Reply:
Oops, now I understand what you just said. Just keep it as is!